### PR TITLE
Game refactor: reduce game.cpp size by moving objects in the right places

### PR DESCRIPTION
**Pull request by nerzhul**
## Part 1: move some input handlings from core to InputHandler
* make InputHandler manage more inputs instead of game

Joystick input is a RealInputHandler only usage, make it intelligent and handle the joystick with keyboard direct.
This permits to remove many getters in game which should be owned by RealInputHandler

## Part 4: own + helpers in InputHandler

### CODE REVIEW

**Comment by rubenwardy** [comment]
It would also be good to move inputhandler from game.h to client/inputhandler.h, and all the other classes to client/inputhandlerimpls.h or something. Ideally game should not include or have access to joystickcontroller etc

**Comment by nerzhul** [comment]
@rubenwardy yeah it's the idea, i just need more time to continue :)

**Comment by nerzhul** [comment]
@rubenwardy it's already the case in fact :)

**Comment by nerzhul** [comment]
@rubenwardy @SmallJoker please review. I removed the WIP flag

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -9,5 +9,6 @@ set(gui_SRCS
 	${CMAKE_CURRENT_SOURCE_DIR}/guiTable.cpp
 	${CMAKE_CURRENT_SOURCE_DIR}/guiVolumeChange.cpp
 	${CMAKE_CURRENT_SOURCE_DIR}/intlGUIEditBox.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/profilergraph.cpp
```
Comment (SmallJoker) [inline src/gui/CMakeLists.txt]: Also add the file to our Android build source list: `build/android/jni/Android.mk`

File: src/gui/guiFormSpecMenu.cpp [hunk 2]
```diff
@@ -129,6 +131,28 @@ GUIFormSpecMenu::~GUIFormSpecMenu()
 	delete m_text_dst;
 }
 
+void GUIFormSpecMenu::create(GUIFormSpecMenu **cur_formspec, Client *client,
+	JoystickController *joystick, IFormSource *fs_src, TextDest *txt_dest)
```
Comment (SmallJoker) [inline src/gui/guiFormSpecMenu.cpp]: Does not belong to moving but as you're touching these lines: could you please use `GUIFormSpecMenu *&cur_formspec`?
-> `cur_formspec->doPause = false` etc.

File: src/gui/CMakeLists.txt (hunk 1 above)
Comment (nerzhul) [reply src/gui/CMakeLists.txt]: done

**Comment by nerzhul** [comment]
Please don't squash on merge

**Comment by nerzhul** [comment]
@SmallJoker it's now fixed

**Pull request merged**